src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable no-console */
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import cn from 'classnames';
+import { DispatchContext } from '../../store/store';
+import { Todo } from '../../types/Todo';
+
+import './TodoItem.scss';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem:React.FC<Props> = ({ todo }) => {
+  const dispatch = useContext(DispatchContext);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const inputEditRef = useRef<HTMLInputElement>(null);
+
+  const toggleTodo = (todoId: number) => {
+    dispatch({ type: 'toggleTodo', payload: todoId });
+  };
+
+  const deleteTodo = (todoId: number) => {
+    dispatch({ type: 'deleteTodo', payload: todoId });
+  };
+
+  const updateTodo = () => {
+    if (!tempTodo) {
+      return;
+    }
+
+    const isEqualTitles = (todo.title === tempTodo.title);
+    const isEmptyTitle = (tempTodo.title === '');
+
+    if (!isEqualTitles && !isEmptyTitle) {
+      dispatch({ type: 'updateTodo', payload: tempTodo });
+    }
+
+    if (isEmptyTitle) {
+      dispatch({ type: 'deleteTodo', payload: tempTodo.id });
+    }
+
+    setTempTodo(null);
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.currentTarget.value.trim();
+    const newTodo = { ...tempTodo, title: newTitle } as Todo;
+
+    setTempTodo(newTodo);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      updateTodo();
+    }
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setTempTodo(null);
+    }
+  };
+
+  const handleOnBlur = () => {
+    updateTodo();
+  };
+
+  useEffect(() => {
+    inputEditRef.current?.focus();
+  }, [tempTodo?.id]);
+
+  return (
+    <li className={cn(
+      { completed: todo.completed },
+      { editing: todo.id === tempTodo?.id },
+    )}
+    >
+      <div className="view">
+        <input
+          type="checkbox"
+          className="toggle"
+          id="toggle-view"
+          checked={todo.completed}
+          onChange={() => toggleTodo(todo.id)}
+        />
+        <label
+          htmlFor="toggle-view"
```

Comment:
please check such behavor.
click on any todo change status for first todo.
looks like it is because of every TodoItem input has not unique id
![chrome-capture-2024-1-12](https://github.com/mate-academy/react_todo-app/assets/33029864/4e5a4648-4666-432c-919c-0079062c5ee3)